addons/website/views/website_templates.xml

```diff
@@ -131,7 +131,7 @@
             <t t-foreach="frontend_languages" t-as="lg">
         <link rel="alternate" t-att-hreflang="lg['hreflang']" t-att-href="url_localized(lang_code=lg['code'], prefetch_langs=True, canonical_domain=canonical_domain)"/>
             </t>
-        <link rel="alternate" hreflang="x-default" t-att-href="url_localized(lang_code=website.default_lang_id._get_cached('code'), canonical_domain=canonical_domain)"/>
+        <link rel="alternate" hreflang="x-default" t-att-href="url_localized(lang_code=website.default_lang_id._get_data(code='code'), canonical_domain=canonical_domain)"/>
```

Comment:
This is wrong, it doesn't return the correct x-default lang: 
![image](https://github.com/odoo/odoo/assets/30048408/9e704479-d88e-4f4d-9721-013689a34881)

Here the website default lang is english but it shows french. I guess it should pass the code of `website.default_lang_id` to `_get_data()`?

Reply:
Good catch. BTW can I use `website._get_cached('default_lang_id')` here?

Reply:
Yes